### ISSUE TITLE
git lfs

### ISSUE THREAD

**Issue by gedw99** [npub1zenn0/go-git-nostr#1]
I wonder if nostr can handle binary images ?

so it can handle git lfs...

**Comment by npub1zenn0** [comment]
Oh yeah, that's something I haven't thought of. Good thing you bring it up.

https://github.com/nostr-protocol/nips/pull/337 and https://github.com/nostr-protocol/nips/pull/337 seem relevant at glance. I think they're about to use bittorrent, not sure how well that will work out for a git patch, something that you'd normally just drive-by download.

**Comment by gedw99** [comment]
I just use the local FS or minio / S3.

works great with git and the golang git lfs plugin.

Why not just let the relay use s3 / minio and the end points can use local FS or minio too ?

Is there at architectural reason we have to use bit torrent ?

**Comment by npub1zenn0** [comment]
Most relays don't expose blob storage afaik. I assume most people don't have access to a global minio/S3 instance that they can push to. So if you're sending the text patch to nostr for free, there's nothing to push a file blob to.

**Comment by gedw99** [comment]
Right .. 

So the relay is just a rendezvous point ?? 

Soa binary only needs to be there until everyone has synced and gotten it onto their local ?

**Comment by npub1zenn0** [comment]
My knowledge is limited here so I may be wrong, but my understanding is  that the relay does store text events, as in it stores "notes", but I'm guessing if you try to upload a big file/event then most relay implementations will reject or drop it very quick. So for text it's like a db but big blobs it's indeed a rendezvous point.

However, I didn't read the NIPs in question in detail yet, I would hope the relays will also host the blobs for at least some time. At a glance it isn't even talked about in the documents.

**Comment by gedw99** [comment]
Thank - I need to read more

**Comment by frbitten** [comment]
Hello
Just to help understanding NOSTR is a communication protocol, it does not define storage. Each relay is free to choose its way of storing events. Currently most use Postgree or SQLite.

For your doubts, the NIP-95 (https://github.com/nostr-protocol/nips/pull/345) can be used, which is about sending blobs in base64 format.

**Comment by gedw99** [comment]
Thankyou @frbitten

**Comment by npub1zenn0** [comment]
I had some more time to think about this one, and I realize that fetching the blobs might not be that essential.

If this tool is used to push the files into nostr, then some existing git host app (e.g. gitea) can pull that from nostr and serve it to users via a regular `git pull`.

**Comment by gedw99** [comment]
https://github.com/icyphox/legit Will do it 

it’s a git server and web gui without the baggage of gitea .

We can also have a nice ssh experience with it via: 

https://github.com/charmbracelet/soft-serve/issues/215

**Comment by gedw99** [comment]
There’s an issue to add git lfs to legit 👍 


https://github.com/icyphox/legit/issues/22

**Comment by npub1zenn0** [comment]
Yeah any frontend can be used, people can pick whatever service they want the commits to get pushed into from nostr. You could find mirrors of the same repo served on many frontends then. In theory.

**Comment by gedw99** [comment]
True .. just wanted to offer options for others. I have found legit to be great
